ui/src/shared/reducers/helpers/fields.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash'
+
+// fieldWalk traverses fields rescursively into args mapping fn on every
+// field
+export const fieldWalk = (fields, fn, acc = []) =>
+  _.compact(
+    _.flattenDeep(
+      fields.reduce((a, f) => {
+        return [...a, fn(f), fieldWalk(_.get(f, 'args', []), fn, acc)]
+      }, acc)
+    )
+  )
+
+// functions returns all top-level fields with type
+export const ofType = (fields, type) =>
+  _.filter(fields, f => _.get(f, 'type') === type)
+
+// functions returns all top-level functions in fields
+export const functions = fields => ofType(fields, 'func')
```

Comment:
Can we group this with the fields function? And have them follow the same naming convention